server/src/app.ts

```diff
@@ -22,83 +24,170 @@ app.get("/alive", (req: Request, res: Response) => {
+          const document = await collection.findOne({
+            _id: new ObjectId(req.params.id),
+          });
+          res.status(200).json(document);
+        } catch (error) {
+          console.error("Error getting item:", error);
+          res.status(500).json({ error: "Failed to get items" });
+        }
+      } else {
+        res.status(500).json({ error: "Not valid document ID" });
+      }
+    });
+
+    //DELETE
+
+    app.delete("/api/v1/items/:id", async (req: Request, res: Response) => {
+      if (ObjectId.isValid(req.params.id)) {
+        try {
+          const collection: Collection = db.collection("recipes");
+          const document = await collection.deleteOne({
+            _id: new ObjectId(req.params.id),
+          });
+          res.status(200).json(document);
+        } catch (error) {
+          console.error("Error deleting item:", error);
+          res.status(500).json({ error: "Failed to delete item" });
+        }
+      } else {
+        res.status(500).json({ error: "Not valid document ID" });
+      }
+    });
+
+    // PATCH
+
+    app.patch("/api/v1/items/:id", async (req: Request, res: Response) => {
+      if (ObjectId.isValid(req.params.id)) {
+        try {
+          const collection: Collection = db.collection("recipes");
+          const existingItem = await collection.findOne({
+            _id: new ObjectId(req.params.id),
+          });
+
+          if (!existingItem) {
+            return res.status(404).json({ error: "Recipe not found" });
+          }
+
+          const {
+            title,
+            description,
+            categories,
+            ingredients,
+            steps,
+          }: {
+            title?: string;
+            description?: string;
+            categories?: string[];
+            ingredients?: string[];
+            steps?: string[];
+          } = req.body;
+
+          const updatedFields: { [key: string]: any } = {};
+          if (title) {
+            updatedFields.title = title;
+          }
+          if (description) {
+            updatedFields.description = description;
+          }
+          if (categories) {
+            updatedFields.categories = categories;
+          }
+          if (ingredients) {
+            updatedFields.ingredients = ingredients;
+          }
+          if (steps) {
+            updatedFields.steps = steps;
+          }
+
+          const updateResult = await collection.updateOne(
+            { _id: new ObjectId(req.params.id) },
+            { $set: updatedFields }
+          );
+
+          res.status(200).json(updateResult);
+        } catch (error) {
+          console.error("Error updating item:", error);
+          res.status(500).json({ error: "Failed to update items" });
+        }
+      } else {
+        res.status(400).json({ error: "Not valid document ID" });
+      }
+    });
+  } catch (error) {
+    console.error("Error connecting to the database:", error);
```

Comment:
if there's an error connecting to db you can do `process.exit(1)` to indicate that process exited with an error